play/src/front/Chat/Connection/Matrix/MatrixChatConnection.ts

```diff
@@ -717,14 +772,39 @@ export class MatrixChatConnection implements ChatConnectionInterface {
     }
 
     clearListener() {
+        this.roomList.forEach((room) => {
+            this.roomList.delete(room.id);
+        });
         this.client?.off(ClientEvent.Room, this.handleRoom);
         this.client?.off(ClientEvent.DeleteRoom, this.handleDeleteRoom);
         this.client?.off(RoomEvent.MyMembership, this.handleMyMembership);
         this.client?.off("RoomState.events" as EmittedEvents, this.handleRoomStateEvent);
         this.client?.off(RoomEvent.Name, this.handleName);
+        this.client?.off(RoomMemberEvent.Typing, this.handleRoomMemberTyping);
     }
-
     async destroy(): Promise<void> {
         await this.client?.logout(true);
     }
 }
+
+function AutoDestroyingMapStoreProxy<K, V extends Required<{ destroy: () => void }>>(mapStore: MapStore<K, V>) {
```

Comment:
Instead of a proxy, you can simply extend the `MapStore` and override the `delete` method :)